assets/mariadbapp/main.go

```diff
@@ -0,0 +1,98 @@
+/*
+   Copyright 2020 SUSE
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/cloudfoundry-community/go-cfenv"
+	"github.com/go-sql-driver/mysql"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func main() {
+	serviceName := os.Getenv("SERVICE_NAME")
+	if serviceName == "" {
+		log.Fatal("SERVICE_NAME not set")
+	}
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	appEnv, err := cfenv.Current()
+	if err != nil {
+		log.Fatal(err)
+	}
+	mariadb, err := appEnv.Services.WithName(serviceName)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	spec, ok := mariadb.Credentials["uri"].(string)
+	if !ok {
+		log.Fatal(fmt.Errorf("URI not supplied"))
+	}
+	uri, err := url.Parse(spec)
+	if err != nil {
+		log.Fatal(err)
+	}
+	config := mysql.NewConfig()
+	config.User = uri.User.String()
+	config.Net = "tcp"
+	config.Addr = uri.Hostname()
+	config.DBName = strings.TrimPrefix(uri.Path, "/")
+	uriStr := config.FormatDSN()
+	fmt.Printf("Connecting to %q\n", uriStr)
+
+	db, err := sql.Open("mysql", uriStr)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
```

Comment:
We already have a config, so avoiding re-parse is good…
```suggestion
	conn, err := mysql.NewConnector(config)
	if err != nil {
		log.Fatal(err)
	}
	db := sql.OpenDB(conn)
	defer db.Close()
```

Reply:
Sounds good, doesn't work.